js/event.ts

```diff
@@ -0,0 +1,242 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import * as domTypes from "./dom_types";
+
+const EVENT_ATTRIBUTES = new WeakMap;
+
+export enum EventPhase {
```

Comment:
Because of some of the challenges building the type library, where our implementations leak extra types and modules that really shouldn't be part of the main Deno runtime library, the more we keep the "surface" APIs that we are trying to adhere to in something like `dom_types.ts`, the less likely we will run into issues with building the type library in the future.  

But who would be consuming this anyways at runtime?